packages/devextreme-react/src/core/extension-component-func.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import {
+  useImperativeHandle,
+  forwardRef,
+  useRef,
+  useEffect,
+  useCallback,
+  ReactElement,
+  memo,
+} from 'react';
+
+import { IHtmlOptions, ComponentBaseRef, ComponentBase } from './component-base-func';
+import { ComponentProps } from './component-func';
+
+type NestedOptionElement = ReactElement<any, React.JSXElementConstructor<any> & { isExtensionComponent: boolean }>;
```

Comment:
Previously, we detected `ExtensionComponent` descendants just by checking their prototype (see the `isPrototypeOf` usage in `component.ts`).

Now, `Validator` component will just use `ExtensionComponent` as its render results, with no way to tell this by the `ReactElement`. So, as long as we already have to assign properties to functional components (see https://github.com/DevExpress/DevExtreme/pull/26783/files#r1506083444), we may want to assign a flag property to all the extension components to make them identifiable by their React elements.
An example of how this flag is used is here: https://github.com/DevExpress/DevExtreme/pull/26783/files#diff-c2ab09c6d3ab5deec02d484b3733fdb2e341a2a66f8f0a63b0b08003df89ea85R23